### ISSUE TITLE
Why generate extra files in a separate package?

### ISSUE THREAD

**Issue by pcj** [tcncloud/protoc-gen-persist#71]
Given a proto file `foo.proto` with go_package `github.com/foo/bar`, the code generator outputs these files under bazel and rules_protobuf (a repo I maintain):

```
bazel-genfiles/github.com/foo/bar/persist_lib/pkg_level_definitions.persist.go
bazel-genfiles/github.com/foo/bar/persist_lib/foo_query_handlers.persist.go
bazel-genfiles/github.com/foo/bar/persist_lib/foo_queries.persist.go
bazel-genfiles/github.com/foo/bar/foo.persist.go
```

Bazel takes a fairly granulated approach to compilation, so typically these generated source code outputs are packaged up into it's own compilation to a single library like `foo_proto.a` (I believe via "go tool pack").  Ultimately, this package gets incorporated via the go linker when producing a final executable  binary file.  

Basically, bazel go tooling does not really like compiling multiple packages simultaneously.  `go build` does not either, but I believe it does a dependency sort internally as part of its build process, if you look under the hood.

The problem here is that the `foo.persist.go` file has a compile dependency on the `persist_lib` package, necessitating partitioning those output files into two groups, compiling just the persist_lib first, then compiling the foo.persist.go file second with a dependency on the persist_lib.a.

Anyway, I'm wondering, is there a need for the files to be emitted in separate packages?  Why not emit all files into the same package?  Would you be open to an option that accomplishes this?

**Comment by namtzigla** [comment]
Initially everything was generated into a single package but we introduced features like type mapping and before and after hooks and we've been forced to split the implementation in multiple packages in order to avoid cyclic  dependencies and we don't  want to use reflection.

For the moment this is the only solution that cover most of our use cases. 
We are thinking to use type aliases feature that has been released in 1.9 but we didn't test it yet.
